### PR TITLE
scriptnode lite crash investigation

### PR DESCRIPTION
**Pull request by zeffii**
this starts a snlite tidy up

### CODE REVIEW

File: nodes/generator/script1_lite.py [hunk 1]
```diff
@@ -324,6 +320,18 @@ def process(self):
...
+        self.n_id = node_id(self)
+        return self.node_dict.get(self.n_id)
+
+    def reset_node_dict(self):
+        self.node_dict[self.n_id] = {}
+
+    def set_node_dict_keys(self, **kwargs):
+        ND = self.node_dict.get(self.n_id)
+        for k, v in kwargs.items():
+            ND[k] = v
```
Comment (Durman) [inline nodes/generator/script1_lite.py:333]: What about using of standard Python dictionary API?
```suggestion
    @property
    def node_dict(self):
        self.n_id = node_id(self)
        return self._node_dict.get(self.n_id)
```

```Python
self.node_dict.clear()
self.node_dict['key'] = my_value
```

Comment (Durman) [reply nodes/generator/script1_lite.py:333]: Node_dict can use `defaultdict` if needed.

```Python
class Node:
    _node_dict = defaultdict(dict)
```

https://docs.python.org/3/library/collections.html#defaultdict-examples

Comment (zeffii) [reply nodes/generator/script1_lite.py:333]: i like to make small changes first.

**Comment by zeffii** [comment]
you can ignore this PR @Durman . it's going to move slowly.

File: nodes/generator/script1_lite.py (hunk 1 above)
Comment (zeffii) [reply nodes/generator/script1_lite.py:333]: what you propose there is in retrospect maybe preferable,  and i will address this later. it looks better, but for now i'm looking for a bug in the prop_search

**Comment by zeffii** [comment]
using the Text pointerProperty does appear to solve this crashing problem, it is needed anyway. I do not have time at the moment to fix all the places that would be effected by switching from StringProperty (script_name) to PointerProperty (script_pointer).

**Pull request closed without merging**